hub: add sync example (#6108)
tldr-pages/tldr: pages/common/hub.md

> If set up as instructed by `hub alias`, one can use `git` to run `hub` commands.
> More information: <https://hub.github.com>.

> - Clone a repository, using its GitHub owner (can be omited if you own it) and the repository name:

Reviewer: How about ?
```suggestion
- Clone a repository using its slug (owners can omit the username):
```
We recently used `slug` in a similar page: [git-fork](https://github.com/tldr-pages/tldr/blob/main/pages/common/git-fork.md)

Author: I don't know this word (I am not a native English speaker :innocent:). Is is slang? Could not find the meaning in [wordreference](https://www.wordreference.com/definition/slug). :thinking: 
Ah, OK, I found this comment: https://github.com/tldr-pages/tldr/pull/6013#discussion_r637399196.
Personally, I prefer using simple words that Atlassian jargon (is it?).

Reviewer 2: It's not Atlassian-specific — it's a common word used for this type of thing (`user/repo`) 🤷🏻

Reviewer 3: I'm also not totally convinced by the word itself, but the example is quite self-exlanatory and this way the user even learns a new word.

Author: OK, I am convinced now.

I have found the word `slug` in various tech documentations:
* BitBucket: https://support.atlassian.com/bitbucket-cloud/docs/what-is-a-slug/
* Azure DevOps: https://docs.microsoft.com/en-us/appcenter/distribution/vsts-deploy
* GitLab: https://docs.gitlab.com/ee/user/project/working_with_projects.html

etc.

Sorry but the initial "here is the Atlassian link" did not "sell" me the word. 😉

Author: Thank you @mfrw, @bl-ue and @marchersimon! I did the change in `fixup` commit c7726572 (in order to preserve the two commits where merging 🤓).

Reviewer: @nicokosi we generally do a `squash & merge` :)

Author: I know but:
> If instead the PR author took the time to craft individual, informative messages for each commit, then use the Rebase and merge method, to honor that work and preserve the history of the changes.

See https://github.com/tldr-pages/tldr/blob/main/contributing-guides/maintainers-guide.md#ii-handling-prs :innocent:

Author: I took that time. :nerd_face:

- Upload the current (local-only) repository to your GitHub account:

`hub create`

> - Fetch git objects from upstream and update local branches:

Author: Copy/pasted from manual (`hub help sync`):
```
hub-sync(1)                                                                            hub manual                                                                           hub-sync(1)

NAME
       hub-sync - Fetch git objects from upstream and update local branches.

SYNOPSIS
       hub sync [--color]

       o If the local branch is outdated, fast-forward it;
       o If the local branch contains unpushed work, warn about it;
       o If the branch seems merged and its upstream branch was deleted, delete it.

       If a local branch does not have any upstream configuration, but has a same-named branch on the remote, treat that as its upstream branch.

OPTIONS
       --color[=WHEN]
           Enable colored output even if stdout is not a terminal. WHEN can be one of "always" (default for --color), "never", or "auto" (default).

SEE ALSO
       hub(1), git-fetch(1)

hub version 2.14.2                                                                    05 Mar 2020                                                                           hub-sync(1)
```

> - Fetch git objects from upstream and update local branches:

Reviewer 3: ```suggestion
- Fetch Git objects from upstream and update local branches:
```

Author: OK. We may open a hub PR! (since I copied/pasted it 😉)

Author: Thank you  @marchersimon. Done in `fixup` commit 2edb1eef.

Reviewer: > OK. We may open a `hub` PR! (since I copied/pasted it) 😉

let's go for it :) .. if there are any discrepancies there
